recipes/finemap/meta.yaml

```
@@ -1,37 +1,57 @@
-# This file created by conda-build 3.22.0
-# meta.yaml template originally from:
-# /home/conda/conda-forge/recipes/finemap, last modified Thu Oct 20 22:53:08 2022
-# ------------------------------------------------
+{% set name = "finemap" %}
+{% set version = "1.4.2" %}
 
 package:
-  name: finemap
-  version: 1.4.1
+  name: {{ name }}
+  version: {{ version }}
+
 source:
-  sha256: 75c919d9cc981bc08e1a982b1632110109ffe6a057e803c9aa36c143aa56d546
-  url: http://www.christianbenner.com/finemap_v1.4.1_x86_64.tgz
+  - url: http://www.christianbenner.com/finemap_v{{ version }}_x86_64.tgz
+    sha256: 3b1fc6eb3c2ccafd647b32e02d0244495cd0ade9ed7d474606c31ebf6e98b0c9
+
 build:
-  binary_relocation: false
-  noarch: generic
-  number: '0'
+  number: 0
+  # skip osx since binary cannot be relinked b/c it was built without the -headerpad_max_install_names option
+  skip: True # [not linux64]
+  run_exports:
+    - {{ pin_subpackage('finemap', max_pin="x") }}
   script:
-    - install -Dd ${PREFIX}/bin
-    - install -D -m 755 finemap_v1.4.1_x86_64 ${PREFIX}/bin/finemap
-  string: '0'
+    - mkdir -p ${PREFIX}/bin
+    - install -m 755 finemap_v{{ version }}_x86_64 ${PREFIX}/bin/finemap
+
 requirements:
-  build: []
-  host: []
-  run: []
+  build:
+    - {{ stdlib("c") }}
+    - libgomp
```

please use the `compiler` directive here.

---

Hi @bgruening ,

Is this what you're thinking? 

```suggestion
    - {{ compiler('cxx') }}
    - libgomp
```

I actually had something like this in f9bee23, but we decided to remove it in https://github.com/bioconda/bioconda-recipes/pull/51978#discussion_r1835275055, since I'm simply copying the binary and not building it. Shall I add it back?

---

Yes this is what I was thinking. If @martin-g is ok with the way you have it now let's keep it. 

Adding libgomp and libgcc will be done by the complier directive and it's a bit more portable. But you eitherway skip everything but linux64. So 🤷

---

I'll take a closer look later today!

---

Do you need anything in the `requirements > build` section if you don't build it from source ?

---

The tests still seem to pass when I remove the `requirements > build` section, but the build will issue a bunch of warnings, which I was concerned about.
```
WARNING (finemap,bin/finemap): /lib64/libdl.so.2 not found in packages, sysroot(s) nor the missing_dso_whitelist... is this binary repackaging?
WARNING (finemap,bin/finemap): /lib64/libm.so.6 not found in packages, sysroot(s) nor the missing_dso_whitelist... is this binary repackaging?
WARNING (finemap,bin/finemap): $RPATH/libgomp.so.1 not found in packages, sysroot(s) nor the missing_dso_whitelist... is this binary repackaging?
WARNING (finemap,bin/finemap): /lib64/libgcc_s.so.1 not found in packages, sysroot(s) nor the missing_dso_whitelist... is this binary repackaging?
WARNING (finemap,bin/finemap): /lib64/libpthread.so.0 not found in packages, sysroot(s) nor the missing_dso_whitelist... is this binary repackaging?
WARNING (finemap,bin/finemap): /lib64/libc.so.6 not found in packages, sysroot(s) nor the missing_dso_whitelist... is this binary repackaging?
WARNING (finemap,bin/finemap): /lib64/ld-linux-x86-64.so.2 not found in packages, sysroot(s) nor the missing_dso_whitelist... is this binary repackaging?
```

---

I don't have any experience with this kind of problems but I think you need to move those deps to `requirements > host`

---

That makes sense. From what I've read online, shared libraries like those are supposed to be placed in the host section to ensure portability. 

But when I move those requirements into the host section, I still get the same warnings mentioned above. So I guess I'll add them to `missing_dso_whitelist`